src/plugins/presentation_util/public/services/README.mdx

```diff
@@ -0,0 +1,145 @@
+---
+id: presentationUtilServices
+slug: /kibana-dev-docs/presentationPlugin/services
+title: Plugin Services Toolkit
+summary: A collection of objects for creating services local to a plugin.
+date: 2020-01-12
+tags: ['kibana', 'presentation', 'services']
+related: ['presentationUtilPlugin']
+---
+
+While Kibana provides a `useKibana` hook for use in a plugin, the number services it provides is very large.  This presents a number of difficulties:
+
+- a direct dependency upon the Kibana environment;
+- a requirement to mock the full Kibana environment when testing or using Storybook;
+- a lack of knowledge as to what services are being consumed at any given time.
+
+To mitigate these difficulties, the Presentation Team creates services within the plugin that then consume Kibana-provided (or other) services.  This is a toolkit for creating simple services within a plugin.
+
+## Overview
+
+- A `PluginServiceFactory` is a function that will return a set of functions-- which comprise a `Service`-- given a set of parameters.
+- A `PluginServiceProvider` is an object that use a factory to start, stop or provide a `Service`.
+- A `PluginServiceRegistry` is a collection of providers for a given environment, (e.g. Kibana, Jest, Storybook, stub, etc).
```

Comment:
Is there a use case for creating multiple instances at the same time? I can't think of a situation where you would need access to both the Kibana one and the jest one at the same time.

Reply:
Yeah, it's unlikely.  The latest version acts upon a single `PluginService` instance.